chia/full_node/block_store.py

```diff
@@ -543,21 +690,42 @@ async def get_block_records_in_range(
         if self.db_wrapper.db_version == 2:
             async with self.db_wrapper.reader_no_transaction() as conn:
                 async with conn.execute(
-                    "SELECT header_hash, block_record FROM full_blocks WHERE height >= ? AND height <= ?",
+                    "SELECT header_hash, block_record,block FROM full_blocks WHERE height >= ? AND height <= ?",
                     (start, stop),
                 ) as cursor:
                     for row in await cursor.fetchall():
                         header_hash = bytes32(row[0])
-                        ret[header_hash] = BlockRecord.from_bytes(row[1])
-
+                        block_record_db: BlockRecordDB = BlockRecordDB.from_bytes(row[1])
+                        plot_filter_info: PlotFilterInfo = plot_filter_info_from_block(zstd.decompress(row[2]))
+                        block_record = block_record_db.to_block_record(
+                            plot_filter_info.pos_ss_cc_challenge_hash,
+                            plot_filter_info.cc_sp_hash,
+                        )
+                        ret[header_hash] = block_record
         else:
-            formatted_str = f"SELECT header_hash, block from block_records WHERE height >= {start} and height <= {stop}"
+            formatted_str = f"""
+                SELECT
+                    block_records.header_hash AS header_hash_br,
+                    full_blocks.header_hash AS header_hash_fb,
+                    block_records.block AS block_br,
+                    full_blocks.block AS block_fb
+                FROM block_records INNER JOIN full_blocks ON block_records.header_hash = full_blocks.header_hash
+                WHERE
+                    block_records.height >= {start} and block_records.height <= {stop}
+                    and full_blocks.height >= {start} and full_blocks.height <= {stop}
+            """
```

Comment:
I imagine combining these queries into one provide a speed-up compared to making two separate queries (as you do above). Is there a reason to not do the same in `get_block_record()`? We probably don't care enough about v1 performance.